bundle/edu.gemini.pot/src/test/scala/edu/gemini/spModel/target/env/GuideGroupSpec.scala

```diff
@@ -183,84 +181,150 @@ class GuideGroupSpec extends Specification with ScalaCheck with Arbitraries {
+        val expectedOrder = focus :: (rights ++ lefts.reverse)
+        val actualOrder   = unrollGroupOptsList(gg)
+        expectedOrder === actualOrder
       }
+  }
 
-    "" in
+  "GuideGroup getAllMatching" should {
+    "partition the collections of guide probe targets by guide probe type" in
       forAll { (g: GuideGroup) =>
-        true
+        val ptns = g.getReferencedGuiders.asScala.map(gp => g.getAllMatching(gp.getType).asScalaList.toSet).toSet
+        (ptns.toList.map(_.length).sum == g.getAll.size) && ptns.forall(p => ptns.forall(q => p == q || p.intersect(q).isEmpty))
       }
+  }
 
-    "" in
+  "GuideGroup cloneTargets" should {
+    "create a new GuideGroup where the underlying ITargets are equal for each guide probe and in the same order" in
       forAll { (g: GuideGroup) =>
-        true
+        val cg  = g.cloneTargets
+        val ts  = g.getAll.asScalaList
+        val cts = cg.getAll.asScalaList
+        (ts.length == cts.length) && ts.zip(cts).forall {
+          case (t1, t2) => t1.getGuider == t2.getGuider && t1.getTargets.asScala.map(_.getTarget) == t2.getTargets.asScala.map(_.getTarget)
+        }
       }
+  }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
+  "GuideGroup getAllContaining" should {
+    "return a subset of guide probe targets containing a specific target for manual groups" in
+    forAll { (g: GuideGroup, t: SPTarget) =>
+      val newGpt = g.getAll.asScalaList.zipWithIndex.map { case (gpt, idx) => if (idx % 2 == 0) gpt else gpt.update(OptionsList.UpdateOps.append(t)) }
+      g.setAll(newGpt.asImList).getAllContaining(t).asScalaList == (g.grp match {
+        case _: ManualGroup    => newGpt.zipWithIndex.collect { case (gpt, idx) if idx % 2 == 1 => gpt }
+        case _: AutomaticGroup => Nil
+      })
+    }
   }
-  */
+
+  "GuideGroup getParamSet" should {
+    "produce a ParamSet that can be read via fromParamSet to result in an equivalent guide group" in
+    forAll { (g: GuideGroup) =>
+      equalGuideGroups(g, GuideGroup.fromParamSet(g.getParamSet(new PioXmlFactory())))
+    }
+  }
+
+  // TODO: Restore this test case once serialization is achieved.
+//  "GuideGroup" should {
+//    "properly serialize and deserialize" in
+//    forAll { (g: GuideGroup) =>
+//      val bao = new ByteArrayOutputStream()
+//      val oos = new ObjectOutputStream(bao)
+//      oos.writeObject(g)
+//      oos.close()
+//
+//      val ois = new ObjectInputStream(new ByteArrayInputStream(bao.toByteArray))
+//      ois.readObject() match {
+//        case g2: GuideGroup => equalGuideGroups(g, g2)
+//        case _              => false
+//      }
+//    }
+//  }
 }
 
 object GuideGroupSpec {
   val AllProbes: List[GuideProbe] = GuideProbeMap.instance.values.asScala.toList.sorted
 
+  def equalGuideGroups(gg1: GuideGroup, gg2: GuideGroup): Boolean = {
+    implicit val EqualSPTarget: Equal[SPTarget] = Equal.equal((t1,t2) => t1.getTarget.getName === t2.getTarget.getName)
```

Comment:
I think this is quite fine for our purposes here.  Might be worthy of a comment though.